api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -512,13 +517,41 @@ public void setDosing(DosingInstructions di) {
 	 */
 	@Override
 	public boolean hasSameOrderableAs(Order otherOrder) {
-		if (!super.hasSameOrderableAs(otherOrder)) {
-			return false;
-		}
-		if (!(otherOrder instanceof DrugOrder)) {
-			return false;
-		}
-		DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
-		return OpenmrsUtil.nullSafeEquals(this.getDrug(), otherDrugOrder.getDrug());
+        if (!super.hasSameOrderableAs(otherOrder)) {
+            return false;
+        }
+        if (!(otherOrder instanceof DrugOrder)) {
+            return false;
+        }
+        DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
+
+        if (isNonCodedDrug() || otherDrugOrder.isNonCodedDrug()) {
```

Comment:
Anyways, it everything is passing we can leave it this way hoping the validator to catch it